library/ssl_tls13_generic.c

```diff
@@ -1144,7 +1578,18 @@ static int ssl_tls13_finalize_change_cipher_spec( mbedtls_ssl_context* ssl )
                 mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_HELLO );
                 break;
             case MBEDTLS_SSL_CLIENT_CCS_AFTER_SERVER_FINISHED:
-                mbedtls_ssl_handshake_set_state( ssl, MBEDTLS_SSL_CLIENT_FINISHED );
+#if defined(MBEDTLS_KEY_EXCHANGE_WITH_CERT_ENABLED)
+                if( ssl->handshake->client_auth )
+                {
+                    mbedtls_ssl_handshake_set_state( ssl,
```

Comment:
On the other hand here, just go to the MBEDTLS_SSL_CLIENT_CERTIFICATE state in all cases.